templates/geosafe/analysis/dynamic_scripts/create/main.html

```diff
@@ -356,6 +357,36 @@
 
     }
 
+    /**
+     * Calculate map's max zoom
+     *
+     * @param data
+     * @param hazard_exposure_group
+     *
+     */
+    function get_max_zoom(data, hazard_exposure_group) {
+        var current_bounds = hazard_exposure_group.getBounds();
+        var corner1 = L.latLng(data['layer_bbox_y1'], data['layer_bbox_x1']);
+        var corner2 = L.latLng(data['layer_bbox_y0'], data['layer_bbox_x0']);
+        var layer_bounds = L.latLngBounds(corner1, corner2);
+
+        var new_sw = [
+            Math.min(
+                current_bounds.getSouthWest().lat,
+                layer_bounds.getSouthWest().lat),
+            Math.min(
+                current_bounds.getSouthWest().lng,
+                layer_bounds.getSouthWest().lng)];
+        var new_ne = [
+            Math.max(
+                current_bounds.getNorthEast().lat,
+                layer_bounds.getNorthEast().lat),
+            Math.max(
+                current_bounds.getNorthEast().lng,
+                layer_bounds.getNorthEast().lng)];
+        map.fitBounds([new_sw, new_ne]);
```

Comment:
Just directly fitBounds to hazard_exposure_group.getBounds() with padding options.

Call this when hazard_exposure_group contains all the layers (last call).